cmd/protoc-gen-jsonshim/jsonshim/plugin.go

```diff
@@ -67,14 +67,19 @@ func (p *Plugin) Generate(file *generator.FileDescriptor) {
 	unmarshalerName := generator.FileName(file) + "Unmarshaler"
 	for _, message := range file.Messages() {
 		// check to make sure something was generated for this type
-		if !gogoproto.HasTypeDecl(file.FileDescriptorProto, message.DescriptorProto) || len(message.GetOneofDecl()) == 0 {
+		if !gogoproto.HasTypeDecl(file.FileDescriptorProto, message.DescriptorProto) {
```

Comment:
Does this mean we will generate Marshal/Unmarshal for every proto type instead of just the top level message?

Reply:
Yes. The original gogo proto descriptors do not provide a clean way of identifying the relationships between two messages. So I opted to generate for all messages.

Reply:
That might be a bit expensive. I might be wrong here but with this approach all the marshalers/unmarshalers for each nested type will be called even though just the top level is sufficient.

Reply:
okay. I see what you mean. Added a condition to not generate nested messages if they don't have oneof fields.

Reply:
Will the child marshallers get called from pbjson, i.e. does pbjson make use of the standard json Un/Marshal interface?